From 2d84d446cf638339e8a2fece4190eaa168cb7968 Mon Sep 17 00:00:00 2001
From: Fernando Guzman Lugo <x0095840@ti.com>
Date: Sun, 18 Apr 2010 00:32:01 -0500
Subject: [PATCH 033/135] DSPBRIDGE: Fix declaration and initialization of sync objects

This patch fixes declaration of completions for sync in
drv_interface and also initialize sync->done object in
msg_sm.c

Signed-off-by: Fernando Guzman Lugo <x0095840@ti.com>
Acked-by: Deepak Chitriki <deepak.chitriki@ti.com>
---
 drivers/dsp/bridge/rmgr/drv_interface.c |    4 ++--
 drivers/dsp/bridge/wmd/msg_sm.c         |    6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/dsp/bridge/rmgr/drv_interface.c b/drivers/dsp/bridge/rmgr/drv_interface.c
index 61aa13a..bd693cd 100644
--- a/drivers/dsp/bridge/rmgr/drv_interface.c
+++ b/drivers/dsp/bridge/rmgr/drv_interface.c
@@ -96,8 +96,8 @@ static int tc_wordswapon;	/* Default value is always false */
 static atomic_t bridge_cref;	/* number of bridge open handles */
 static struct workqueue_struct *bridge_rec_queue;
 static struct work_struct bridge_recovery_work;
-static DECLARE_COMPLETION_ONSTACK(bridge_comp);
-static DECLARE_COMPLETION_ONSTACK(bridge_open_comp);
+static DECLARE_COMPLETION(bridge_comp);
+static DECLARE_COMPLETION(bridge_open_comp);
 static bool recover;
 #endif
 
diff --git a/drivers/dsp/bridge/wmd/msg_sm.c b/drivers/dsp/bridge/wmd/msg_sm.c
index dae6e67..9e04a35 100644
--- a/drivers/dsp/bridge/wmd/msg_sm.c
+++ b/drivers/dsp/bridge/wmd/msg_sm.c
@@ -192,8 +192,8 @@ dsp_status bridge_msg_create_queue(struct msg_mgr *hmsg_mgr,
 	if (DSP_SUCCEEDED(status)) {
 		msg_q->sync_done = kzalloc(sizeof(struct sync_object),
 							GFP_KERNEL);
-		if (msg_q->sync_event)
-			sync_init_event(msg_q->sync_event);
+		if (msg_q->sync_done)
+			sync_init_event(msg_q->sync_done);
 		else
 			status = DSP_EMEMORY;
 	}
@@ -201,7 +201,7 @@ dsp_status bridge_msg_create_queue(struct msg_mgr *hmsg_mgr,
 	if (DSP_SUCCEEDED(status)) {
 		msg_q->sync_done_ack = kzalloc(sizeof(struct sync_object),
 							GFP_KERNEL);
-		if (msg_q->sync_event)
+		if (msg_q->sync_done_ack)
 			sync_init_event(msg_q->sync_done_ack);
 		else
 			status = DSP_EMEMORY;
-- 
1.7.0.4

